pkg/backup/backupdest/backup_index.go

```diff
@@ -31,11 +35,25 @@ import (
 // information.
 func WriteBackupIndexMetadata(
 	ctx context.Context,
+	execCfg *sql.ExecutorConfig,
 	user username.SQLUsername,
 	makeExternalStorageFromURI cloud.ExternalStorageFromURIFactory,
 	details jobspb.BackupDetails,
 ) error {
-	ctx, sp := tracing.ChildSpan(ctx, "backupdest.WriteBackupIndexMetadata")
+	indexStore, err := makeExternalStorageFromURI(
```

Comment:
I think we are leaking this cloud storage instance and we need to defer a Close.